fpmsyncd/routesync.cpp

```diff
@@ -30,6 +54,458 @@ RouteSync::RouteSync(RedisPipeline *pipeline) :
+                    break;
+
+                len -= NLMSG_ALIGN(rtnh->rtnh_len);
+                rtnh = RTNH_NEXT(rtnh);
+            }			
+        }
+    }
+    return true;
+}
+
+void RouteSync::onEvpnRouteMsg(struct nlmsghdr *h, int len)
+{
+    struct rtmsg *rtm;
+    struct rtattr *tb[RTA_MAX + 1];
+    void *dest = NULL;
+    char anyaddr[16] = {0};
+    char dstaddr[16] = {0};
+    int  dst_len = 0;
+    char buf[MAX_ADDR_SIZE];
+    char destipprefix[IFNAMSIZ + MAX_ADDR_SIZE + 2] = {0};
+    int nlmsg_type = h->nlmsg_type;
+    unsigned int vrf_index;
+
+    rtm = (struct rtmsg *)NLMSG_DATA(h);
+
+    /* Parse attributes and extract fields of interest. */
+    memset(tb, 0, sizeof(tb));
+    netlink_parse_rtattr(tb, RTA_MAX, RTM_RTA(rtm), len);
+
+    if (tb[RTA_DST])
+        dest = RTA_DATA(tb[RTA_DST]);
+    else
+        dest = anyaddr;
+
+    if (rtm->rtm_family == AF_INET)
+    {
+        if (rtm->rtm_dst_len > IPV4_MAX_BITLEN)
+        {
+            return;
+        }
+        memcpy(dstaddr, dest, IPV4_MAX_BYTE);
+        dst_len = rtm->rtm_dst_len;
+    }
+    else if (rtm->rtm_family == AF_INET6)
+    {
+        if (rtm->rtm_dst_len > IPV6_MAX_BITLEN) 
+        {
+            return;
+        }
+        memcpy(dstaddr, dest, IPV6_MAX_BYTE);
+        dst_len = rtm->rtm_dst_len;
+    }
+
+    SWSS_LOG_DEBUG("Rx MsgType:%d Family:%d Prefix:%s/%d", h->nlmsg_type, rtm->rtm_family,
+                    inet_ntop(rtm->rtm_family, dstaddr, buf, MAX_ADDR_SIZE), dst_len);
+
+    /* Table corresponding to route. */
+    if (tb[RTA_TABLE])
+        vrf_index = *(int *)RTA_DATA(tb[RTA_TABLE]);
+    else
+        vrf_index = rtm->rtm_table;
+
+    if (vrf_index)
+    {
+        if (!getIfName(vrf_index, destipprefix, IFNAMSIZ))
+        {
+            SWSS_LOG_ERROR("Fail to get the VRF name (ifindex %u)", vrf_index);
+            return;
+        }
+        /*
+         * Now vrf device name is required to start with VRF_PREFIX,
+         * it is difficult to split vrf_name:ipv6_addr.
+         */
+        if (memcmp(destipprefix, VRF_PREFIX, strlen(VRF_PREFIX)))
+        {
+            SWSS_LOG_ERROR("Invalid VRF name %s (ifindex %u)", destipprefix, vrf_index);
+            return;
+        }
+        destipprefix[strlen(destipprefix)] = ':';
+    }
+
+    /* Full mask route append prefix length, or else resync cannot match. */
+    snprintf(destipprefix + strlen(destipprefix), sizeof(destipprefix) - strlen(destipprefix), "%s/%u", 
+                inet_ntop(rtm->rtm_family, dstaddr, buf, MAX_ADDR_SIZE), dst_len);
+
+    SWSS_LOG_DEBUG("Receive route message dest ip prefix: %s Op:%s", 
+                    destipprefix,
+                    nlmsg_type == RTM_NEWROUTE ? "add":"del");
+
+    /*
+     * Upon arrival of a delete msg we could either push the change right away,
+     * or we could opt to defer it if we are going through a warm-reboot cycle.
+     */
+    bool warmRestartInProgress = m_warmStartHelper.inProgress();
```

Comment:
I'm not sure I understand it here. Why do we want to check for warmboot status in this fpmsyncd flow? How do we hit this case?

Reply:
This code is similar to onMsg processing(existing code to add/del prefix), when warmStart is in progress then don't delete the entry from the table.